examples/platforms/nrf528xx/src/radio.c

```diff
@@ -695,9 +706,12 @@ otError otPlatRadioGetTransmitPower(otInstance *aInstance, int8_t *aPower)
 otError otPlatRadioSetTransmitPower(otInstance *aInstance, int8_t aPower)
 {
     OT_UNUSED_VARIABLE(aInstance);
+    uint8_t channel         = nrf_802154_channel_get();
+    int8_t  channelMaxPower = nrf5GetChannelMaxTransmitPower(channel);
 
-    sDefaultTxPower = aPower;
-    nrf_802154_tx_power_set(aPower);
+    sDefaultTxPower   = aPower;
+    sPowerInitialized = true;
+    nrf_802154_tx_power_set(aPower < channelMaxPower ? aPower : channelMaxPower);
```

Comment:
The function `otPlatRadioGetTransmitPower` calls `nrf_802154_tx_power_get` to get the transmit power, which may cause that when we are calling the function `otPlatRadioSetTransmitPower `to set the output power to A, but the fcuntion `otPlatRadioGetTransmitPower` returns the output power as B. Should we consider this case?

Reply:
This is currently the intended behavior. All the power will be guarded with the max power table.

Reply:
Done